vernac/comFixpoint.ml

```diff
@@ -255,7 +255,7 @@ let declare_fixpoint local poly ((fixnames,fixdefs,fixtypes),pl,ctx,fiximps) ind
       Some (List.map (Option.cata (EConstr.of_constr %> Tactics.exact_no_check) Tacticals.New.tclIDTAC)
         fixdefs) in
     let evd = Evd.from_ctx ctx in
-    Lemmas.start_proof_with_initialization (Global,poly,DefinitionBody Fixpoint)
+    Lemmas.start_proof_with_initialization ~allow_nested:true (Global,poly,DefinitionBody Fixpoint)
```

Comment:
I haven't fully understood what `declare_fixpoint` does but it may well be that this should be set to `false` instead.

Reply:
I think this is what happens for `Fixpoint foo (n:nat) : True.` so it should allow nested iff the option is on.